main.cpp

```diff
@@ -310,162 +339,217 @@ struct Dishwasher
     void cleanDishes(Dishes dishes);
     void dryDishes(bool shouldDryDishes = true, int temp = 55);
     void selfClean(int afterRun = 45);
+
+    Dishes dishes;
 };
+Dishwasher::Dishwasher()
+{
+    capacityPlates = 15;
+    capacityCupsGlasses = 18;
+    capacityCutlery = 30;
+    timePerRun = 135;
+    waterConsumption = 30.5;
+}
+Dishwasher::Dishes::Dishes()
+{
+    oiliness = 25.76f;
+    soupBowls = 4;
+    regularPlates = 12;
+    cups = 8;
+    glasses = 5;
+}
 void Dishwasher::Dishes::clogDrain(std::string residues)
 {
     if(residues == "too much")
     {
-        std::cout << "clogged";
+        std::cout << "clogged" << std::endl;
     }
 }
 void Dishwasher::Dishes::tarnish(bool isSensitive)
 {
     if(isSensitive)
     {
-        std::cout << "cutlery successfully tarnished";
+        std::cout << "cutlery successfully tarnished" << std::endl;
     }
 }
 void Dishwasher::Dishes::shatter(bool isPiledBadly)
 {
     if(isPiledBadly)
     {
-        std::cout << "let's break some stuff";
+        std::cout << "let's break some stuff" << std::endl;
     }
 }
-void Dishwasher::cleanDishes(Dishwasher::Dishes dishes)
+void Dishwasher::cleanDishes(Dishwasher::Dishes newDishes)
 {
-    std::cout << "cleaning dishes...";
-    dishes.oilyness = 0.f;
+    if(dishes.oiliness > 0)
+    {
+        std::cout << "dishes are dirty, oiliness: " << dishes.oiliness << std::endl;
+        std::cout << "cleaning dishes..." << std::endl;
+        newDishes.oiliness = 0.f;
+        dishes.oiliness = newDishes.oiliness;
+        std::cout << "dishes cleaned, oiliness: " << dishes.oiliness << std::endl;
+    }
```

Comment:
you should be checking the oiliness of the `newDishes`, not the member variable `dishes`
after making this revision, you should copy all of the properties from `newDishes` into `dishes`.
```
if( newDishes.oiliness > 0 )
{
    std::cout << "dishes are dirty, oiliness: " << newDishes.oiliness << std::endl;
    std::cout << "cleaning dishes..." << std::endl;
    
    //now modify any properties of the 'newDishes' object that indicates how dirty it is
    //like you're doing on this line:
    newDishes.oiliness = 0.f;
    
    //now copy the properties from 'newDishes' to 'dishes, like you're doing on this line:
    dishes.oiliness = newDishes.oiliness;
    
    //then print out that you're done: 
    std::cout << "dishes cleaned, oiliness: " << dishes.oiliness << std::endl;
}
```
- [ ] revise